### ISSUE TITLE
filter returns empty data frame with no columns if result is empty

### ISSUE THREAD

**Issue by marvinwu** [Gmousse/dataframe-js#57]
```
const DataFrame = require('dataframe-js').DataFrame

const df = new DataFrame([
    { c1: 1, c2: 6 },
    { c1: 1, c3: 2 }
], ['c1', 'c2', 'c3', 'c4'])
df.filter(row => row.get('c1') > 1).show()
```
output:
|  |
---

now it returns a empty df with no columns, will it be better to return an df with original columns but no rows ? 

ie: returns
| c1        | c2        | c3        | c4        |
------------------------------------------------

it creates problem when the df later do joining with other dataframe if no column df is returned

**Comment by Gmousse** [comment]
Hi,
Indeed it should returns an empty dataframe with the same schema.

I will fix it

**Comment by Gmousse** [comment]
Fixed, it will be merged in 1.4.0.

Feel free to test it via #72
